lib/src/geo/crs.dart

```diff
@@ -53,32 +58,39 @@ abstract class Crs {
 }
 
 @immutable
-abstract class _CrsWithStaticTransformation extends Crs {
+abstract class CrsWithStaticTransformation extends Crs {
```

Comment:
Maybe we should comment what a CRS with a static transform is (or more interestingly, exemplify when the transform is non-static).

I don't know off-hand when that occurs. One thing I could possibly think of is compensating for tectonic shift? Say you have a statue in your garden pinned with a lat/lon, and 10 years down the line, your coordinate is still pointing to the same place on Earth, but the tectonic plate has moved, effectively changing the location of your statue.

I think _dynamic_ CRS like EPSG:3857 and EPSG:4326 need a _coordinate epoch_, a timestamp associated with every coordinate, to be able to compensate for the tectonic shift that has occurred. The movement is calculated relative to a _frame reference epoch_, where tectonic plate locations/velocities are known. The movements are small (cm-level per year) and only relevant for fairly precise measurements, or old measurements relative to the frame reference.

Some local CRS are referred to as _static_ because they are _plate-fixed_, i.e. move with the tectonic plates, in which case you don't need to compensate.

It doesn't seem this is supported by `Proj4Dart` though.

Reply:
Also, an easy way to go from Proj4-string to a custom `CrsWithStaticTransformation` would be very nice. My CRS is not known at compile time:
```dart
Proj4Crs.fromFactory(
  code: 'EPSG:xyz',
  proj4Projection: proj4.Projection.add(
      'EPSG:xyz',
      proj4Str,
    ),
);
```

Reply:
Don't forget to compensate for relativistic contractions and elongations based on the reference frame of the observer :)

Jokes aside, I feel that you're reading too much into the name. I don't think there was ever an intend to support time evolution.
To counter your point, the only non-static-transform CRS is the Proj4Crs. Maybe I chose the name (not sure) to contrast the fact that Proj4Crs supports an array of transformations depending on the zoom level. Arguably "Static" might not be the best name. Either way I'm only changing visibility here.

> Also, an easy way to go from Proj4-string to a custom CrsWithStaticTransformation would be very nice

I don't see why you wouldn't be able to implement your own Proj4 Crs or extend the existing one (though the existing one isn't a CrsWithStaticTransform but I also don't see why that would matter).

Reply:
> Either way I'm only changing visibility here.

Yes, in exposing it I figured it could be more clearly documented than something strictly internal (although for internal use it could also help, since the name is a bit confusing).

Reply:
Fair. What name would you suggest?

(I'm also open to other suggestions. I'm only exposing it to monomorphize some code. We could also drop this optimization or maybe go some other path. Open to suggestions)

Reply:
> Fair. What name would you suggest?

I don't know tbh, still haven't fully understood the difference.

> (I'm also open to other suggestions. I'm only exposing it to monomorphize some code. We could also drop this optimization or maybe go some other path. Open to suggestions)

I think a broader approach extending beyond a single layer could be beneficial for the handling of points, and that might warrant some careful consideration, as opposed to adding a new point type for a specific case as part of addressing something else.

> Don't forget to compensate for relativistic contractions and elongations based on the reference frame of the observer :)

It's a increasing in popularity, especially for regions like Australia that move relatively fast 😄

Reply:
> It's a increasing in popularity, especially for regions like Australia that move relatively fast 😄

Them Aussies, maybe we should flip their latitudes :).

> I think a broader approach extending beyond a single layer could be beneficial for the handling of points, and that might warrant some careful consideration, as opposed to adding a new point type for a specific case as part of addressing something else.

Are we mixing conversations? Where do you see the connection between the:
 * CRS base class name
 * and/or a singular fixed transformation for e.g. Epsg3857
 * and internal point type
?

Reply:
What I think I inferred is that this conversation has fully moved away from "CrsWithStaticTransformation" and is now fully about the point type?

To additionally throw some assumptions in there:  "you care about the internal types because you're already planning to open up these internal layers in your fork"?

Otherwise...

> So cache them to your heart's content, but that isn't necessarily related to introducing a new point type to monomorphize the code, or how are the two inseparable?

All of this is separable, that's why it felt like this conversation is slipping between unrelated topics.

> What I'm trying to say is, it's a bit of a local hack, even if it's an internal point. It could be worth keeping math.Point until we have a better/broader option. That doesn't imply there will be a better option, but it keeps things a bit more streamlined.

But we're already mixing, e.g. we're using record types in some places. These are internal APIs, we should do what's best for our users (keeping our code in a maintainable state is clearly also good for the users).

> If math.Point is a bottleneck it should probably be seen as such in a broader context than just a single layer.

I don't understand the "if". Yes, math.Point is a bottleneck. I'd love to get rid of it, however changing public APIs and thus forcing every user to migrate is a very different story than optimizing our code paths internally.

Reply:
Will mark this conversation as resolved based on you feedback. Please re-open if you feel the issue has not been addressed.